github/VisualStudio
# Added details about when the artifacts are uploaded (#1) (#2)

* Handle IO and json exceptions gracefully

Fixes #516

* Use https version of script submodule url

* Add a loader for external packages

Add a way to load external packages that are included in the vsix and
expose the IUIProvider as a global service, so those packages can make
use of our services.
Load the StartPage package if we're running on Dev15 Preview 4

* Revert "Use https version of script submodule url"

No, bad commit is bad, breaks CI.
This reverts commit d9d394d15f5a25375be258733caa0e6a65e8fcf4.

* Disable non-public stuff without breaking public builds

* Let's compare in insensitive ways

* Be explicit about GUIDs

* No racing here thank you

* Bump StartPage dependency

* sigh

* Fix the obvious brain fart

* No, really, what's up with my brain today

* Release builds of the StartPage package, v1.0

* Bump version to 2.0.14.6

* Make sure we don't deadlock loading services

* Apparently the vs gallery refuses to parse the Express_All key even though VS is fine with it

* Remove extra targets from vsix completely

* Bump version to 2.0.15.0

* Bump version to 2.0.15.1

* Update octokit to 0.22.0

This fixes #534 via https://github.com/octokit/octokit.net/pull/1411. Previously the request to /site/sha to detect an enterprise instance wasn't getting sent due to octokit's implementation of `RedirectHandler`.

* Bump version to 2.0.15.2

* Change repository model class hierarchy.

In preparation for the following rename:

- SimpleRepositoryModel -> LocalRepositoryModel
- RepositoryModel -> RemoteRepositoryModel

Change the repository model hierarchy to make RepositoryModel and
SimpleRepositoryModel inherit from a common base rather than
RepositoryModel from SimpleRepositoryModel. This means that information
that is not relevant to a remote repository (local path etc) is no
longer present on RepositoryModel.

* Renamed repository model classes/interfaces.

- IRepositoryModelBase -> IRepositoryModel
- SimpleRepositoryModel -> LocalRepositoryModel
- RepositoryModel -> RemoteRepositoryModel

* Addressed PR comments.

* Don't allow null/empty repository name.

* Fix Guard.ArgumentNotEmptyString

I was calling length on a possibly null string.

* Don't allow null local repository path.

* Make CloneUrl setter protected.

* Added doc comments.

* use ProjectItem instead of ActiveDocument name

* Add more tests to check casing

* Fix refresh of PR list switching to "create" form.

Fixes #480. Issue was happening because of an extra PRCreation
item in the nav stack when Refresh was clicked 2nd time.

It was there because when navigating forward for the 2nd Create,
GoForward method wasn't removing everything after the current item.

* Removed TempFileBaseClass.

Instead tests should construct a new instance of
`TestBaseClass.TempDirectory` and use that. Fixes #543.

* Removed debug code.

* Make things compile.

* Removed duplicate Guard class.

* Fix potential NullReferenceException.

ArgumentNotEmptyString should handle null value being passed in.

* Added unit tests for Guard,

* Bump version to 2.0.15.3

* Rename Simple -> LocalRepositoryModel elsewhere.

There were a few places that referred to SimpleRepositoryModel by name
rather than symbol that needed to be changed to LocalRepositoryModel.

* CloneUrl can be null but Owner should never be.

* Add unit test for PR refresh issue.

* Use new ILocalRepositoryModel.

* Added TempRepository helper class.

* add async method GetLatestSha

* update Setup Repository to mock new method

* make tests pass

* Allow canonical refs in SetTrackingBranch

Allow canonical or friendly branch refs to be passed to
SetTrackingBranch.

* Fix incorrect doc comments.

* Added Fetch method that accepts refspecs.

To GitClient.

* Removed private.

* return the head Sha when path is null

* use the local path instead of returning the head sha

* Convert GitClient methods to use Tasks.

* Run git tasks on b/g thread.

* document method, and add null check

* update documentation and handle when repo is null

* Clean up publish code

* Let's not block!

* Code review changes:

* Remove InternalsVisibleTo, make PkgCmdIDList public for UnitTests to be able to use it.
* Remove explicit "private" access modifiers.
* Remove #regions.
* Follow lambda convention.
* Remove TODO comment that didn't totally make sense.

* Add unit tests for clone dialog search box cursor issue.

* Fix disable of filterbox when loading repos list on RepositoryClone dialog.
- Filterbox will be disabled if the loading fails, otherwise it will always be enabled.

* Added unit test for disabling filter textbox when no repositories where found.

* Disable filter textbox when no loading has completed but no repositories where found.

* start Pull request section of manifest

* include more pr steps

* change text

* Change link to button

* Corrected logic for filterTextEnabled observable.

* Fixed test which was not initializing VM.

* more pr instructions and test, started open/copy link and gist steps

* removed bracket around single parameter lambda.

* swap steps and complete gist steps

* add more specific instructions

* start page instructions

* add buton style

* Fix submodule url to be public (fixes #571)

* Making sure RaisePropertyChanged is called

* Adding test to show TrackingCollection updating equal object

* Making the fix to TrackingCollection

* Fixing other TrackingCollection unit tests

* Updating comment

* Correcting test OrderByMatchesOriginalOrder

* Make ScrollViewer hack reusable.

WPF's ScrollViewer is broken, so we need to hack around it. We already
had this hack in GitHubConnectContent, so moved the hack to a central
place that can be used wherever it's needed.

* Grammar nazi on the prowl

* Gist Creation Notification

* Removing unintentional import statement
